src/main/java/net/openhft/chronicle/wire/Wires.java

```diff
@@ -399,6 +400,10 @@ public static Wire acquireBinaryWire() {
         return wire;
     }
 
+    public static ScopedResource<Wire> acquireBinaryWireScoped() {
+        return WireInternal.BINARY_WIRE_SCOPED_TL.get();
+    }
+
     @NotNull
     public static Bytes<?> acquireAnotherBytes() {
         if (Jvm.isDebug())
```

Comment:
applying this new approach uniformly will enable us to remove this different behaviour when running in debugger. @peter-lawrey I know you did this deliberately - there are comments like "when in debug, the output becomes confused if you reuse the buffer" but it seems sub-optimal to change behaviour of code when running in the debugger, and you would hope that this new approach would ensure that no buffers will get erroneously reused

Reply:
It's sub-optimal when running debugging but more optimal when not running debugging. This change would allow us to remove such checks. The small overhead isn't a problem is rarely used.